### ISSUE TITLE
pressure state for incompressible media

### ISSUE THREAD

**Issue by Mathadon** [iea-annex60/modelica-annex60#225]
When using incompressible media derivatives of pressures are generated when ```massDynamics not = steadyState``` because of the ```stateSelect.prefer``` declaration. This causes additional derivatives to be computed since there is no real state for the pressure when the medium is incompressible.

I will do some tests to check if it's best to remove the stateSelect.prefer or to set it to the mixing volume mass. We should also run unit tests and performance benchmarks to assess the impact.

**Comment by mwetter** [comment]
@Mathadon : I looked into that as on branch `issue_313` of the `Buildings` library, the command 
```
translateModel("Buildings.Fluid.FMI.Examples.FMUs.MixingVolume");
```
fails with 
```
The model requires derivatives of some inputs as listed below:
1 inlet.p
```
Can you point to a test case where you see these additional derivatives? Simply setting
```
  redeclare replaceable model extends BaseProperties(
    Xi(each stateSelect=if preferredMediumStates then StateSelect.prefer else StateSelect.default),
    final standardOrderComponents=true) "Base properties"
``` 
in `Media.Air` is not sufficient to make the translation succeed. Without having a test case, I don't quite understand your issue above.

**Comment by Mathadon** [comment]
If you enable ```Output information when differentiating for index reduction``` you get:

```
Differentiated the equation
vol.dynBal.medium.p = inlet.p;

giving
der(vol.dynBal.medium.p) = der(inlet.p);

Differentiated the equation
0.8333333333333334*vol.dynBal.m = vol.dynBal.medium.p/101325;

giving
0.8333333333333334*der(vol.dynBal.m) = 9.869232667160129E-06*der(vol.dynBal.medium.p);

Differentiated the equation
vol.dynBal.medium.p = inlet.p;

giving
der(vol.dynBal.medium.p) = der(inlet.p);

Differentiated the equation
0.8333333333333334*vol.dynBal.m = vol.dynBal.medium.p/101325;

giving
0.8333333333333334*der(vol.dynBal.m) = 9.869232667160129E-06*der(vol.dynBal.medium.p);

Differentiated the equation
vol.dynBal.mXi[1] = vol.dynBal.m*vol.dynBal.medium.Xi[1];

giving
der(vol.dynBal.mXi[1]) = der(vol.dynBal.m)*vol.dynBal.medium.Xi[1]+ vol.dynBal.m*der(vol.dynBal.medium.Xi[1]);
```
The inlet pressure is identical to the pressure state value. A discontinuous change in inlet pressure would therefore lead to an infinite mass flow rate. For solving the problem correctly Dymola therefore requires the derivative of the inlet pressure to be known, since ```m_flow = f(der(m)) = f(der(p))```. If you remove the pressure state by setting ```massDynamics = steadyState``` the problem disappears. Dymola still outputs:
```
Differentiated the equation
vol.dynBal.medium.p = inlet.p;

giving
der(vol.dynBal.medium.p) = der(inlet.p);
```
but this derivative is not used in the model so it seems to be no problem. This of course means that massDynamics need to be steady state. If you do want to use compressible air I think you'll need to make FMU's of a series connection of a pressure drop component and a mixing volume. And not of an individual mixing volume.


Note that this problem is not entirely related to the issue. An example of this issue is example ```Buildings.Fluid.MixingVolumes.Examples.MixingVolume```. First change the medium to ```Buildings.Media.Water```. Then you can compare the model with and without ```p(start=p_start, stateSelect=StateSelect.prefer),```. With ```StateSelect.prefer``` the total number of derivatives ```der(*)``` in ```dsmodel.mof``` is 73 instead of 11. For instance you get equations like:
```
der(vol1.dynBal.medium.p) := Buildings.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp:der
        (res11.dp, 1.2649110640673518, 0.6, 1.0, 0.0, 0.0)*der(P.y)/(
        Buildings.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp:der(res12.dp,
         1.2649110640673518, 0.6, 1.0, 0.0, 0.0)+Buildings.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp:der
        (res11.dp, 1.2649110640673518, 0.6, 1.0, 0.0, 0.0));
```
while we're not at all interested in the derivative of the pressure, only in the pressure.

**Comment by mwetter** [comment]
It turns out that https://github.com/iea-annex60/modelica-annex60/commit/116194dddec58d1f5e94e646abcab8a72bf3ccb2 gives smaller systems of equations in some models of the Buildings library. For example, 
for `Buildings.Examples.Tutorial.Boiler.System7`:
```
*** Warning: BoilerSystem7.mat: Found translation statistics for initialization for linear in old but not in new results.
 Old = 11
*** Warning: BoilerSystem7.mat: Translation statistics for simulation and results changed for linear.
 Old = 0, 0, 0, 0, 0, 0, 0, 11
 New = 0, 0, 0, 0, 0, 0, 0
```
and computing time changed from about 10 to 9 seconds.

**Comment by Mathadon** [comment]
This was fixed by #248
